server/src/main/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -1760,8 +1761,22 @@ public void updateSegmentMetadata(final Set<DataSegment> segments)
           @Override
           public Void inTransaction(Handle handle, TransactionStatus transactionStatus) throws Exception
```

Comment:
Nit: Since we are changing this code anyway, see if you can use a lambda here instead.